### PR TITLE
GT 276

### PR DESCRIPTION
**Pull request by dsgoers**

### CODE REVIEW

File: app/controllers/attachments_controller.rb [hunk 1]
```diff
@@ -1,6 +1,12 @@
 # frozen_string_literal: true
 
-class AttachmentsController < SecureController
+class AttachmentsController < ApplicationController
+  before_action :authorize!, only: [:create, :update, :destroy]
+
+  def download
+    redirect_to load_attachment.file.url, status: :found
```
Comment (soberstadt) [inline app/controllers/attachments_controller.rb]: I've never used this found status, do you need it here?

File: app/models/attribute.rb [hunk 2]
```diff
@@ -1,29 +1,25 @@
...
 
   def set_defaults
     self.is_translatable ||= false
   end
 
-  def duplicate_keys # see note on same method in attachment.rb
-    return unless resource.attachments.find_by(key: key).present?
-    raise 'Key is current used by an Attachment.'
+  def lower_key
+    self.key = key.downcase if key.present?
```
Comment (soberstadt) [inline app/models/attribute.rb]: `self.key = key&.downcase` should also work if you want it.

File: app/controllers/attachments_controller.rb (hunk 1 above)
Comment (dsgoers) [inline app/controllers/attachments_controller.rb]: No, but the app developers requested it.

Comment (soberstadt) [inline app/controllers/attachments_controller.rb]: Huh, ok

Comment (dsgoers) [inline app/controllers/attachments_controller.rb]: To differentiate between other types of redirect.

**Pull request merged**